tests/test_video_capturer.py

```diff
@@ -1,14 +1,24 @@
 import time
+import numpy as np
+import mock
+import random
+from mock import patch
 
 from src.jetson.video_capturer import VideoCapturer
+def mock_read():
+    img = np.zeros((300, 300, 3), np.uint8)
+    img[:] = (random.randint(0,255), random.randint(0,255), random.randint(0,255))
```

Comment:
Is there a specific reason this is random?

Reply:
A real call to cv2.VideoCapture.read should produce a different frame every iteration assuming the camera is working. We have a check for this in test_update to make sure different frames are being produced.

Reply:
given this is the case I would like to see that when the camera isn't detecting new frames it properly (no pixels chnaged) it is rebooted or that there is some form of error handling that is done here so us as developers are aware that an issue occurred.

Reply:
cv2.videoCapture will return False or None if there aren't any frames: https://docs.opencv.org/2.4/modules/highgui/doc/reading_and_writing_images_and_video.html

I think there's a check for `if frame is not None` somewhere, so that might be the one you want

Reply:
this is implemented in latest commit